embed-code-go/fragmentation/fragmentation_utils.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024, TeamDev. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+package fragmentation
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode/utf8"
+)
+
+func ShouldFragmentize(file string) bool {
+	info, err := os.Stat(file)
+	if err != nil {
+		panic(err)
+	}
+
+	isFile := !info.IsDir()
+	isValidEncoding := IsValidEncoding(file)
+
+	return isFile && isValidEncoding
+}
+
+func IsFileUTF8Encoded(filename string) (bool, error) {
```

Comment:
Please re-group all these functions, so that they are not making a generic mess.

Clearly, there are several here. I see _at least_ two:

* `io.go`, which are helpers for file-related I/O,
* something like `/fragmentation/lookup.go` (or whatever name you think fits best) — which helps to locate the fragments.

Maybe, there are more.

Each of the helper functions should be documented.